src/adaptors/pancakeswap/index.js

```diff
@@ -0,0 +1,186 @@
+    'https://api.coingecko.com/api/v3/simple/price?ids=pancakeswap-token%2Cbinancecoin%2Cethereum&vs_currencies=usd'
+  );
+
+  const cakePrice = prices['pancakeswap-token'].usd;
+  const ethPrice = prices.ethereum.usd;
+  const bnbPrice = prices.binancecoin.usd;
+
+  return { cakePrice, ethPrice, bnbPrice };
+};
+
+const main = async () => {
+  const { cakePrice, ethPrice, bnbPrice } = await getBaseTokensPrice();
+  const masterChef = new web3.eth.Contract(masterChefABI, MASTERCHEF_ADDRESS);
+  const poolsCount = await masterChef.methods.poolLength().call();
+  const totalAllocPoint = await masterChef.methods
+    .totalRegularAllocPoint()
+    .call();
+  const cakeRateToRegularFarm = await masterChef.methods
+    .cakePerBlock(true)
+    .call();
+  const normalizedCakePerBlock = cakeRateToRegularFarm / 1e18;
+
+  const [poolsRes, lpTokensRes] = await Promise.all(
+    ['poolInfo', 'lpToken'].map((method) =>
+      sdk.api.abi.multiCall({
+        abi: masterChefABI.filter(({ name }) => name === method)[0],
+        calls: [...Array(Number(poolsCount - 1)).keys()].map((i) => ({
+          target: MASTERCHEF_ADDRESS,
+          params: i,
+        })),
+        chain: 'bsc',
+      })
+    )
+  );
+  const poolsInfo = poolsRes.output.map((res) => res.output);
+  const lpTokens = lpTokensRes.output.map((res) => res.output);
+
+  const [reservesRes, supplyRes, masterChefBalancesRes] = await Promise.all(
+    ['getReserves', 'totalSupply', 'balanceOf'].map((method) =>
+      sdk.api.abi.multiCall({
+        abi: lpTokenABI.filter(({ name }) => name === method)[0],
+        calls: lpTokens.map((address) => ({
+          target: address,
+          params: method === 'balanceOf' ? [MASTERCHEF_ADDRESS] : null,
+        })),
+        chain: 'bsc',
+      })
+    )
+  );
+  const reservesData = reservesRes.output.map((res) => res.output);
+  const supplyData = supplyRes.output.map((res) => res.output);
+  const masterChefBalData = masterChefBalancesRes.output.map(
+    (res) => res.output
+  );
+
+  const pools = await Promise.all(
+    poolsInfo.map((pool, i) =>
+      getPairInfo(lpTokens[i]).then(({ pair: pairInfo }) => {
+        // the first two pools are for lotteries, etc.
+        if (i < 2) return;
+
+        const poolInfo = poolsInfo[i];
+        const reserves = reservesData[i];
+
+        const supply = supplyData[i];
+        const masterChefBalance = masterChefBalData[i];
+
+        const reserveUSD = calculateReservesUSD(
+          pairInfo.name,
+          reserves,
+          masterChefBalance / supply,
+          bnbPrice,
+          cakePrice,
+          ethPrice,
+          pairInfo.token0.decimals,
+          pairInfo.token1.decimals
+        )
+          .div(1e18)
+          .toString();
+        const pool = {
+          pool: pairInfo.name,
+          chain: utils.formatChain('binance'),
+          project: 'pancakeswap',
+          symbol: pairInfo.name,
+          tvlUsd: reserveUSD,
+          apy: calculateApy(
```

Comment:
from a brief check on frontend, seems like there is some delta btw adaptor apy and UI eg:

```
  {
    "pool": "USDT-WBNB",
    "chain": "Binance",
    "project": "pancakeswap",
    "symbol": "USDT-WBNB",
    "tvlUsd": "167047154.66379022044044984363",
    "apy": 2.3555793949912465
  },
```

vs:
<img width="1184" alt="Screenshot 2022-07-05 at 11 59 21" src="https://user-images.githubusercontent.com/74533329/177397037-1936a75d-ed82-495f-8408-305f7c0bef35.png">